### ISSUE TITLE
Incorrect "spawned process is never waited on"

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust-clippy#13459]
### Summary

I got this warning since the most recent nightly clippy, but the warning seems to be wrong:

### Lint Name

clippy::zombie_processes

### Reproducer

I tried this code:

```rust
                let mut child = cmd.spawn().expect("failed to spawn process");
                let child_stdin = child.stdin.take().unwrap();
                // Write stdin in a background thread, as it may block.
                let exit_status = thread::scope(|s| {
                    s.spawn(|| {
                        let mut child_stdin = child_stdin;
                        // Ignore failure, it is most likely due to the process having terminated.
                        let _ = child_stdin.write_all(&info.stdin);
                    });
                    child.wait().expect("failed to run command")
                });
```

I saw this happen:

```
warning: spawned process is never `wait()`ed on
   --> cargo-miri/src/phases.rs:669:33
    |
669 |                 let mut child = cmd.spawn().expect("failed to spawn process");
    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: consider calling `.wait()`
    = note: not doing so might leave behind zombie processes
    = note: see https://doc.rust-lang.org/stable/std/process/struct.Child.html#warning
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#zombie_processes
    = note: `#[warn(clippy::zombie_processes)]` on by default
```

I expected to see this happen:

There's a `wait` inside the `scope` closure, so there should be no warning.

The `wait` has to be in that closure to avoid deadlocks: sending the input to the child process must happen concurrently with waiting for the child to finish.

### Version

rustc 1.83.0-nightly (76ed7a1fa 2024-09-26)
binary: rustc
commit-hash: 76ed7a1fa40c3f54d3fd3f834e12bf9c932d0146
commit-date: 2024-09-26
host: x86_64-unknown-linux-gnu
release: 1.83.0-nightly
LLVM version: 19.1.0
